sdk/ml/azure-ai-ml/azure/ai/ml/_azure_environments.py

```diff
@@ -189,3 +196,45 @@ def _resource_to_scopes(resource):
     """
     scope = resource + "/.default"
     return [scope]
+
+def _get_clouds_by_metadata_url(metadata_url, timeout=ArmConstants.DEFAULT_TIMEOUT):
+    """Get all the clouds by the specified metadata url
```

Comment:
nit: this now is really just get cloud (singular) by metadata, but the doc string says get all clouds

Reply:
I think the metadataurl returns multiple clouds, you can see the get_cloud method has to specify the cloud from the list provided by this method

Reply:
ah yeah you're right, forgot about that :D